app/controllers/application_controller.rb

```diff
@@ -2,4 +2,5 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  # before_action :authenticate_user!
```

Comment:
Consider re-enabling this line in `ApplicationController`:
```ruby
before_action :authenticate_user!
```
By requiring users are logged-in in `ApplicationController`in this way, actions in controllers that subclass `ApplicationController` (which is all controllers) will also require users are logged-in.

This design choice is [secure by default](https://en.wikipedia.org/wiki/Secure_by_default). 

Favor secure by default design decisions as they help make an app more secure when we, the developer, are forgetful. (This is preferable to choices that make an app less secure when we're forgetful).

When you want to write a controller action that does not require login, then remember to add a line like the following to the controller in question:

```
class FooController < ApplicationController

  # Allow anonymous (not logged-in) users to request the FooController#index
  # and FooController#show actions. All other actions in this controller will
  # require login:
  skip_before_action :authenticate_user!, only: [:index, :show]
  ...
end
```